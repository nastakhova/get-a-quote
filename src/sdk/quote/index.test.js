import React from 'react';
import { Provider } from 'react-redux'
import renderer from 'react-test-renderer';
import store from '../../store'

import Quote from './index';

it('renders correctly when there are no quotes', () => {
  const tree = renderer.create(
	<Provider store={store}>
        <Quote />
    </Provider>
  	).toJSON();
  expect(tree).toMatchSnapshot();
});

it('renders correctly when there quote', () => {
  const quote = {
        "ID": 1736,
        "title": "Asher Trotter",
        "content": "<p>Creating something out of thin air is easy. It&#8217;s finding the air that&#8217;s hard.</p>\n",
        "link": "https://quotesondesign.com/asher-trotter/"
  };
  const tree = renderer.create(
  	<Provider store={store}>
        <Quote data={quote} />
    </Provider>
  	).toJSON();
  expect(tree).toMatchSnapshot();
});

