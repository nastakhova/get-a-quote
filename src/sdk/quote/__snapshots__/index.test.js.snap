// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly when there are no quotes 1`] = `
<p>
  Uppss... Something goes wrong!
</p>
`;

exports[`renders correctly when there quote 1`] = `
Array [
  <button
    className="button button--remove"
    onClick={[Function]}
  >
    x
  </button>,
  <blockquote
    className="quotes-list__quote"
  >
    <div
      className="quotes-list__content"
      dangerouslySetInnerHTML={
        Object {
          "__html": "<p>Creating something out of thin air is easy. It&#8217;s finding the air that&#8217;s hard.</p>
",
        }
      }
    />
    <div
      className="quotes-list__footer"
    >
      <button
        className="button button--edit"
        onClick={[Function]}
      >
        edit
      </button>
      <cite
        className="quotes-list__cite"
      >
        <a
          href="https://quotesondesign.com/asher-trotter/"
          rel="noopener noreferrer"
          target="_blank"
        >
          Asher Trotter
        </a>
      </cite>
    </div>
  </blockquote>,
]
`;
